src/shared/src/impls.rs

```diff
@@ -164,6 +164,35 @@ impl StoredUserProfile {
         new_profile.updated_timestamp = now;
         Ok(new_profile)
     }
+
+    /// # Errors
+    ///
+    /// Will return Err if there is a version mismatch or if the dApp ID is already hidden.
```

Comment:
Personally I wouldn't return an error if the app is already hidden.  It is often helpful if functions are idenpontent, so saying "hide x!  hide x!  hide x!" should get the answer "ok, ok, ok, calm down!".  But it's a matter of taste.  Up to you.

Reply:
eheh agreed! changed

Reply:
Thanks!